crates/matrix-sdk-common/src/executor.rs

```diff
@@ -50,6 +54,32 @@ where
     JoinHandle { remote_handle, abort_handle }
 }
 
+pub trait BoxFutureExt<'a, T: 'a> {
+    fn box_future(self) -> BoxFuture<'a, T>;
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl<'a, F, T> BoxFutureExt<'a, T> for F
+where
+    F: Future<Output = T> + 'a + Send,
+    T: 'a,
+{
+    fn box_future(self) -> BoxFuture<'a, T> {
+        self.boxed()
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+impl<'a, F, T> BoxFutureExt<'a, T> for F
+where
+    F: Future<Output = T> + 'a,
+    T: 'a,
+{
+    fn box_future(self) -> BoxFuture<'a, T> {
+        self.boxed_local()
+    }
+}
```

Comment:
impl blocks can be merged the same way.

Reply:
I was struggling to get it to compile with the extra `+ Send` trait needed on the not-wasm32 version.